src/SFA.DAS.ApplyService.Database/APR-2311.sql

```diff
@@ -0,0 +1,71 @@
+    [ApplicationId],
+    [FinancialReviewStatus] as Status,                  
+    JSON_VALUE(apply.FinancialGrade, '$.SelectedGrade') AS SelectedGrade,
+	JSON_VALUE(apply.FinancialGrade, '$.FinancialDueDate') AS FinancialDueDate,
+		JSON_VALUE(apply.FinancialGrade, '$.GradedBy') AS GradedBy,	
+		JSON_VALUE(apply.FinancialGrade, '$.GradedDateTime') AS GradedOn,	
+		JSON_VALUE(apply.FinancialGrade, '$.Comments') AS Comments,	
+		JSON_VALUE(apply.FinancialGrade, '$.ExternalComments') AS ExternalComments,	
+        JSON_VALUE(apply.FinancialGrade, '$.FinancialEvidences') AS FinancialEvidences,	
+		JSON_VALUE(apply.FinancialGrade, '$.ClarificationRequestedOn') AS ClarificationRequestedOn,		
+		JSON_VALUE(apply.FinancialGrade, '$.ClarificationRequestedBy') AS ClarificationRequestedBy,		
+		JSON_VALUE(apply.FinancialGrade, '$.ClarificationResponse') AS ClarificationResponse
+  FROM [dbo].[Apply] apply
+  where applicationID not in (select applicationId from FinancialReview)
+  and FinancialGrade is not null
+
+-- there can be up to 4 clarification files
+insert into FinancialReviewClarificationFile (applicationId, Filename)
+SELECT 
+    [ApplicationId],
+		JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[0].Filename') AS Filename  
+  FROM [dbo].[Apply] apply
+  WHERE JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[0].Filename') not in (select filename from FinancialReviewClarificationFile)
+  AND FinancialGrade is not null
+  AND JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[0].Filename') is not null
+
+
+insert into FinancialReviewClarificationFile (applicationId, Filename)
+SELECT 
+    [ApplicationId],
+		JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[1].Filename') AS Filename  
+  FROM [dbo].[Apply] apply
+  WHERE JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[1].Filename') not in (select filename from FinancialReviewClarificationFile)
+  AND FinancialGrade is not null
+  AND JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[1].Filename') is not null
+
+insert into FinancialReviewClarificationFile (applicationId, Filename)
+SELECT 
+    [ApplicationId],
+		JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[2].Filename') AS Filename  
+  FROM [dbo].[Apply] apply
+  WHERE JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[2].Filename') not in (select filename from FinancialReviewClarificationFile)
+  AND FinancialGrade is not null
+  AND JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[2].Filename') is not null
+
+insert into FinancialReviewClarificationFile (applicationId, Filename)
+SELECT 
+    [ApplicationId],
+		JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[3].Filename') AS Filename  
+  FROM [dbo].[Apply] apply
+  WHERE JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[3].Filename') not in (select filename from FinancialReviewClarificationFile)
+  AND FinancialGrade is not null
+  AND JSON_VALUE(apply.FinancialGrade, '$.ClarificationFiles[3].Filename') is not null
```

Comment:
This looks like it'll work. However I'm pretty sure you could do all of this with one Cross Apply statement and it'll be atomic

Reply:
It'll be something like this:

INSERT INTO FinancialReviewClarificationFile (applicationId, Filename)
SELECT apply.ApplicationId, FinancialGrade.Filename 
FROM [dbo].[Apply] apply
  CROSS APPLY OPENJSON(apply.FinancialGrade,'$.ClarificationFiles') WITH (Filename varchar(255) FinancialGrade
WHERE apply.FinancialGrade IS NOT NULL

Reply:
yeah that works.  Vast improvement, thanks